components/vc/shared/wheelSpeed/wheelSpeed.c

```diff
@@ -80,34 +88,62 @@ float32_t wheelSpeed_getSpeedLinear(wheel_E wheel)
     switch (wheelSpeed_config.sensorType[wheel])
     {
         case WS_SENSORTYPE_CAN_RPM:
+        {
+            float32_t temp = 0.0f;
+            if (wheelSpeed_config.config[wheel].rpm(&temp) == CANRX_MESSAGE_VALID)
             {
-                float32_t temp = 0.0f;
-                if (wheelSpeed_config.config[wheel].rpm(&temp) == CANRX_MESSAGE_VALID)
-                {
-                    mps = HZ_TO_MPS(RPM_TO_HZ(temp));
-                }
+                mps = HZ_TO_MPS(RPM_TO_HZ(temp));
             }
-            break;
+        }
+        break;
+
         case WS_SENSORTYPE_CAN_MPS:
+        {
+            float32_t temp = 0.0f;
+            if (wheelSpeed_config.config[wheel].mps(&temp) == CANRX_MESSAGE_VALID)
             {
-                float32_t temp = 0.0f;
-                if (wheelSpeed_config.config[wheel].mps(&temp) == CANRX_MESSAGE_VALID)
-                {
-                    mps = temp;
-                }
+                mps = temp;
             }
-            break;
+        }
+        break;
+
         default:
-            mps = HZ_TO_MPS(wheels.hz[wheel]);
+            mps = HZ_TO_MPS(wheelSpeed_data.wheels.hz[wheel]);
             break;
     }
 
     return mps;
 }
 
+float32_t wheelSpeed_getSlipRatio(void)
+{
+    return wheelSpeed_data.slipRatio;
+}
+
+static void calcSlipRatio_100Hz(void)
+{
+    float32_t FR_wheelSpeed = wheelSpeed_getSpeedRotational(WHEEL_FR);
+    float32_t RR_wheelSpeed = wheelSpeed_getSpeedRotational(WHEEL_RR);
+    float32_t FL_wheelSpeed = wheelSpeed_getSpeedRotational(WHEEL_FL);
+    float32_t RL_wheelSpeed = wheelSpeed_getSpeedRotational(WHEEL_RL);
```

Comment:
IIRC if you want to use this on a given VC, it presently only tracks itself. On VCFRONT you will need to RX VCREAR wheel speeds and then add these to the configuration struct and vice versa for VCREAR